app/mailers/folio/devise_mailer.rb

```diff
@@ -62,17 +74,27 @@ def omniauth_conflict(authentication, opts = {})
 
     initialize_from_record(@record)
 
-    template_data = {
-      USER_CONFLICT_PROVIDER: authentication.human_provider,
-      USER_CONFLICT_RESOLVE_URL: main_app.users_auth_resolve_conflict_url(conflict_token: authentication.conflict_token)
-    }
+    with_locale(@record, opts) do |locale|
+      template_data = {
+        LOCALE: locale,
+        USER_CONFLICT_PROVIDER: authentication.human_provider,
+        USER_CONFLICT_RESOLVE_URL: main_app.users_auth_resolve_conflict_url(conflict_token: authentication.conflict_token)
+      }
 
-    email_template_mail template_data,
-                        headers_for(:omniauth_conflict, opts).merge(subject: t("devise.mailer.omniauth_conflict.subject"),
-                                                                    mailer: "Devise::Mailer")
+      email_template_mail template_data,
+                          headers_for(:omniauth_conflict, opts).merge(subject: t("devise.mailer.omniauth_conflict.subject"),
+                                                                      mailer: "Devise::Mailer")
+    end
   end
 
   private
+    def with_locale(record, opts)
+      locale = opts[:locale] || record.preferred_locale || record.auth_site.locale
+      I18n.with_locale(locale) do
+        yield locale
+      end
+    end
```

Comment:
Proc je tohle jen na devise mailer? To by melo byt nekde obecne, ne? Zaroven bych to prejmenoval na `with_user_locale`, aby se to nepletlo s tim `I18n.with_locale`

Reply:
Určitě přejmenuju, to bude lepší. Na mailer jsem to dal protože se pak v Auctify v jednotlivých mailerech trochu lišilo, jak se má locale získat.

Ale je pravda, že by to mohlo jít zobecnit, kouknu se na to